### PR TITLE
change file_sd names to files, for consistency

### LINKED ISSUE #1682: Make rules_files and file_sd names consistent.
For the rules files we call the config parameter `rules_files`, whereas for file_sd we also have a list of globals to files and call it `names`.

For consistency I propose that we rename the file_sd parameter to `files` or similar.

### PR DESCRIPTION
**Pull request by alileza**
@brian-brazil should all the struct name change into `SDFiles` and `SDFile` also?

Fixes #1682

### CODE REVIEW

**Comment by sdurrheimer** [comment]
This change is odd to me, we lost the naming consistency with other SD mechanism `{type}_sd_configs`.
It's also a breaking change.

File: config/testdata/conf.good.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ scrape_configs:
   # metrics_path defaults to '/metrics'
   # scheme defaults to 'http'.
 
-  file_sd_configs:
+  sd_files:
     - names:
```
Comment (fabxc) [inline config/testdata/conf.good.yml]: I think Brian was referring to the "names" field in the issue rather than "file_sd_configs".

Comment (brian-brazil) [inline config/testdata/conf.good.yml]: Yes I was just talking about `names` as I noticed that inconsistency while giving the workshop.

**Comment by alileza** [comment]
hmm, well i guess i misunderstand the issue. I'll just make the changes then :D

**Comment by fabxc** [comment]
👍 

I don't mind breaking here for consistency. Any objections @juliusv @beorn7?

**Comment by brian-brazil** [comment]
:+1:

**Comment by beorn7** [comment]
I'm not claiming to be the most qualified person in these matters, but I certainly have no objections.

**Comment by fabxc** [comment]
Let's make it so.

**Comment by grobie** [comment]
I'd vote to provide descriptive error messages when renaming/removing config options. The consolidation is great though.

**Comment by juliusv** [comment]
Yeah, just got the first confused user on IRC, since the error message is quite misleading at the moment:

```
15:22 < gouthamve> Hi, was there a breaking change wrt file sd?
15:22 < gouthamve> I just compiled Prometheus on a new machine and the config doesn't work any more
15:22 < gouthamve> ERRO[0000] Couldn't load configuration (-config.file=/etc/config/prometheus/prometheus.yml): file service discovery config must contain at least one path name  
                   source=main.go:218
```

**Comment by juliusv** [comment]
Maybe it would suffice to check the overflow (XXX) fields before doing any other validation? Then this would have at least given an "unknown fields ..." error message.

**Comment by alileza** [comment]
checking XXX fields sound good, i'll do the changes then.

**Pull request merged**